applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp

```diff
@@ -571,20 +572,22 @@ void TrussElement3D2N::CreateTransformationMatrix(
   // local x-axis (e1_local) is the beam axis  (in GID is e3_local)
   double VectorNorm;
   VectorNorm = MathUtils<double>::Norm(direction_vector_x);
-  if (VectorNorm != 0)

```

Comment:
This is not a equality check so the old version might be fine.

Reply:
you are right but the new version should also do the job here I guess

Reply:
👍 but I think `!= 0` cleaner

Reply:
why is this not an equality check?
`VectorNorm` is a double

Reply:
I just don't get the benefit of this change. Maybe you can help me.

The division `a/b` is not defined for `b=0`. If you check for `b != 0` it should be fine.

`std::numeric_limits<T>::epsilon()` is defined as the difference between 1 and the next representable number. Do you want to use the smallest positive non-zero number? Why? That would be `std::numeric_limits<T>::min()`. Even then I don't understand why we should exclude this number. So I would keep things simple.

For the `== 1` comparison: If the intention is to avoid numeric errors I would say that `epsilon` is too small.

Reply:
hm what you say is not wrong for sure

This would also require a global solution eventually

We could discuss it in the next days